internal/adapter/persistentvolumeclaim.go

```diff
@@ -0,0 +1,218 @@
+package adapter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/volume"
+	adaptererr "github.com/portainer/k2d/internal/adapter/errors"
+	k2dtypes "github.com/portainer/k2d/internal/adapter/types"
+	"github.com/portainer/k2d/internal/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+func (adapter *KubeDockerAdapter) CreatePersistentVolumeClaim(ctx context.Context, persistentVolumeClaim *corev1.PersistentVolumeClaim) error {
+	if persistentVolumeClaim.Labels["app.kubernetes.io/managed-by"] == "Helm" {
+		persistentVolumeClaimData, err := json.Marshal(persistentVolumeClaim)
+		if err != nil {
+			return fmt.Errorf("unable to marshal deployment: %w", err)
+		}
+		persistentVolumeClaim.ObjectMeta.Annotations["kubectl.kubernetes.io/last-applied-configuration"] = string(persistentVolumeClaimData)
+	}
+
+	_, err := adapter.cli.VolumeCreate(ctx, volume.CreateOptions{
+		Name:   buildPersistentVolumeName(persistentVolumeClaim.Name, persistentVolumeClaim.Namespace),
+		Driver: "local",
+		Labels: map[string]string{
+			k2dtypes.NamespaceLabelKey:                              persistentVolumeClaim.Namespace,
+			k2dtypes.PersistentVolumeLabelKey:                       buildPersistentVolumeName(persistentVolumeClaim.Name, persistentVolumeClaim.Namespace),
+			k2dtypes.PersistentVolumeClaimLabelKey:                  persistentVolumeClaim.Name,
+			k2dtypes.PersistentVolumeClaimLastAppliedConfigLabelKey: persistentVolumeClaim.ObjectMeta.Annotations["kubectl.kubernetes.io/last-applied-configuration"],
+		},
+		// DriverOpts: map[string]string{
+		// 	"o": "size" + persistentVolumeClaim.Spec.Resources.Requests.Storage().String(),
+		// },
+	})
+
+	if err != nil {
+		return fmt.Errorf("unable to create a Docker volume for the request persistent volume claim: %w", err)
+	}
+
+	return nil
+}
+
+func (adapter *KubeDockerAdapter) DeletePersistentVolumeClaim(ctx context.Context, persistentVolumeClaimName string, namespaceName string) error {
+	persistentVolumeName := buildPersistentVolumeName(persistentVolumeClaimName, namespaceName)
+
+	err := adapter.cli.VolumeRemove(ctx, persistentVolumeName, true)
+	if err != nil {
+		return fmt.Errorf("unable to remove Docker volume: %w", err)
+	}
+
+	return nil
+}
+
+func (adapter *KubeDockerAdapter) GetPersistentVolumeClaim(ctx context.Context, persistentVolumeClaimName string, namespaceName string) (*corev1.PersistentVolumeClaim, error) {
+	labelFilter := filters.NewArgs()
+	labelFilter.Add("name", persistentVolumeClaimName)
+	labelFilter.Add("label", fmt.Sprintf("%s=%s", k2dtypes.NamespaceLabelKey, namespaceName))
```

Comment:
Same here, try to use inspect when looking for a single resource.

Reply:
Done!

Reply:
Actually, switching to volumeIspect causes the following issue. Let's have a chat later tonight:

```
(⎈|k2d:N/A)➜  k2d-pvc (feat/storage) k apply -f tests/pvc.yaml                                                                                                                                                                                           [9:07:14]  ✭ ✱
Error from server (InternalError): error when retrieving current configuration of:
Resource: "/v1, Resource=persistentvolumeclaims", GroupVersionKind: "/v1, Kind=PersistentVolumeClaim"
Name: "node-red-claim", Namespace: "default"
from server for: "tests/pvc.yaml": an error on the server ("unable to get persistent volume claim: unable to list volumes to return the output values from a Docker volume: Error response from daemon: get k2d-pv-default-node-red-claim: no such volume") has prevented the request from succeeding (get persistentvolumeclaims node-red-claim)
```